### PR TITLE
Attempt to fix magnitude calculation by putting everything in Angstroms!

### PR DESCRIPTION
**Pull request by mnicholl**

### CODE REVIEW

File: mosfit/modules/engines/magnetar.py [hunk 1]
```diff
@@ -23,22 +23,24 @@ def process(self, **kwargs):
...
             for x in self._times
         ]
 
-        # print(ts)
-        #
-        # raise SystemExit
-        #
-        luminosities = [Ep / tp / (1. + t * DAY_CGS / tp)**2 for t in ts]
+
+        luminosities = [2 * Ep / tp / (1. + 2 * t * DAY_CGS / tp)**2 for t in ts]
```
Comment (guillochon) [inline mosfit/modules/engines/magnetar.py]: Some `2`s appeared here?

File: mosfit/modules/observables/photometry.py [hunk 2]
```diff
@@ -70,6 +72,8 @@ def __init__(self, **kwargs):
         self._filter_integrals = [0.0] * self._n_bands
         self._average_wavelengths = [0.0] * self._n_bands
         self._band_offsets = [0.0] * self._n_bands
+        FLUX_STD = 3631 * u.Jy.cgs.scale / u.Angstrom.cgs.scale * C_CGS
+        ANG_CGS = u.Angstrom.cgs.scale
```
Comment (guillochon) [inline mosfit/modules/observables/photometry.py]: Move these defs into the class itself (like in `Blackbody`) and refer to them as `self.FLUX_STD` e.g.

File: mosfit/modules/observables/photometry.py [hunk 3]
```diff
@@ -224,6 +230,7 @@ def process(self, **kwargs):
         self._instruments = kwargs['instruments']
         self._bandsets = kwargs['bandsets']
         zp1 = 1.0 + kwargs['redshift']
+        ANG_CGS = u.Angstrom.cgs.scale
```
Comment (guillochon) [inline mosfit/modules/observables/photometry.py]: This isn't necessary if definition is in class itself

File: mosfit/modules/seds/blackbody.py [hunk 4]
```diff
@@ -28,37 +31,51 @@ def process(self, **kwargs):
...
-            radius_phot = self._radius_phot[li]
-            temperature_phot = self._temperature_phot[li]
+                rest_wavs = (self._sample_wavelengths[bi]
+                            * u.Angstrom.cgs.scale / zp1)
+                if not evaled:
+                    sed = ne.evaluate(
+                        'fc * radius_phot**2 / rest_wavs**5 / '
+                        '(exp(xc / rest_wavs / temperature_phot) - 1.0)')
+                    evaled = True
+                else:
```
Comment (guillochon) [inline mosfit/modules/seds/blackbody.py]: This isn't safe now, you have two different numexpr arguments for radio/non-radio, so a re-evaluate will be in error if switching between radio/non-radio observations. The numexpr should be the same for both (so I'd convert to wavelength before passing into numexpr).

Comment (guillochon) [inline mosfit/modules/seds/blackbody.py]: Also this section now duplicates almost two identical blocks of code. Shouldn't be necessary.

File: mosfit/modules/engines/magnetar.py (hunk 1 above)
Comment (mnicholl) [inline mosfit/modules/engines/magnetar.py]: Yeah these 2s are from the original 1971 paper that gave the equation - which is the only place I can verify that the timescale is derived correctly! So the 2s should stay

**Pull request merged**